src/runtime_src/xdp/profile/plugin/aie_debug/aie_debug_plugin.cpp

```diff
@@ -0,0 +1,400 @@
+
+  AIEDebugPlugin::AIEDebugPlugin() 
+      : XDPPlugin()
+  {
+    db->registerPlugin(this);
+    db->registerInfo(info::aie_status);
+    getPollingInterval();
+  }
+
+  AIEDebugPlugin::~AIEDebugPlugin()
+  {
+    // Stop the polling thread
+    endPoll();
+
+    // Write out final version of file and unregister plugin
+    if (VPDatabase::alive()) {
+      for (auto w : writers) {
+        w->write(false);
+      }
+
+      db->unregisterPlugin(this);
+    }
+  }
+
+  // Get polling interval (in usec; no minimum)
+  void AIEDebugPlugin::getPollingInterval()
+  {
+    mPollingInterval = xrt_core::config::get_aie_status_interval_us();
+  }
+
+  // Get tiles to debug
+  void AIEDebugPlugin::getTilesForDebug(void* handle)
+  {
+    std::shared_ptr<xrt_core::device> device = xrt_core::get_userpf_device(handle);
+
+    // Capture all tiles across all graphs
+    // Note: in the future, we could support user-defined tile sets
+    auto graphs = xrt_core::edge::aie::get_graphs(device.get());
+    for (auto& graph : graphs) {
+      mGraphCoreTilesMap[graph] = xrt_core::edge::aie::get_event_tiles(device.get(), graph,
+          xrt_core::edge::aie::module_type::core);
+    }
+
+    // Report tiles (debug only)
+    {
+      std::stringstream msg;
+      msg << "Tiles used for AIE debug:\n";
+      for (const auto& kv : mGraphCoreTilesMap) {
+        msg << kv.first << " : ";
+        for (const auto& tile : kv.second) {
+          msg << "(" << tile.col << "," << tile.row << "), ";
+        }
+        msg << "\n";
+      }
+      xrt_core::message::send(severity_level::debug, "XRT", msg.str());
+    }
+  }
+
+  std::string AIEDebugPlugin::getCoreStatusString(uint32_t status) {
+    std::string statusStr;
+
+    if (status & 0x1)
+      statusStr += "Enable,";
+    if (status & 0x2)
+      statusStr += "Reset,";
+    if (status & 0x4)
+      statusStr += "Memory_Stall_S,";
+    if (status & 0x8)
+      statusStr += "Memory_Stall_W,";
+    if (status & 0x10)
+      statusStr += "Memory_Stall_N,";
+    if (status & 0x20)
+      statusStr += "Memory_Stall_E,";
+    if (status & 0x40)
+      statusStr += "Lock_Stall_S,";
+    if (status & 0x80)
+      statusStr += "Lock_Stall_W,";
+    if (status & 0x100)
+      statusStr += "Lock_Stall_N,";
+    if (status & 0x200)
+      statusStr += "Lock_Stall_E,";
+    if (status & 0x400)
+      statusStr += "Stream_Stall_SS0,";
+    if (status & 0x800)
+      statusStr += "Stream_Stall_SS1,";
+    if (status & 0x1000)
+      statusStr += "Stream_Stall_MS0,";
+    if (status & 0x2000)
+      statusStr += "Stream_Stall_MS1,";
+    if (status & 0x4000)
+      statusStr += "Cascade_Stall_SCD,";
+    if (status & 0x8000)
+      statusStr += "Cascade_Stall_MCD,";
+    if (status & 0x10000)
+      statusStr += "Debug_Halt,";
+    if (status & 0x20000)
+      statusStr += "ECC_Error_Stall,";
+    if (status & 0x40000)
+      statusStr += "ECC_Scrubbing_Stall,";
+
+    return statusStr;
```

Comment:
To help with the readability, you might wish to add the follow code to remove the comma (',').
```
if (!statusStr.empty())
  statusStr.pop_back();
```

Reply:
Yes but then add a comment like
```c++
// Remove the trailing comma
```